pilot/pkg/xds/sds.go

```diff
@@ -78,14 +85,16 @@ func sdsNeedsPush(updates model.XdsUpdates) bool {
 // Invalid resource names are ignored
 func (s *SecretGen) parseResources(names []string, proxy *model.Proxy) []SecretResource {
 	res := make([]SecretResource, 0, len(names))
+	pkpConf := (*mesh.ProxyConfig)(proxy.Metadata.ProxyConfig).GetPrivateKeyProvider()
+	pkpConfHash := xxhashv2.Sum64String(pkpConf.String())
```

Comment:
Please check pkpConf exists before hash

Reply:
@hzxuzhonghu Both `GetPrivateKeyProvider` and `.String()` handles nil input
```
func (x *ProxyConfig) GetPrivateKeyProvider() *PrivateKeyProvider {
	if x != nil {
		return x.PrivateKeyProvider
	}
	return nil
}
```
```
func (o MarshalOptions) Format(m proto.Message) string {
	if m == nil || !m.ProtoReflect().IsValid() {
		return "<nil>" // invalid syntax, but okay since this is for debugging
	}
	o.allowInvalidUTF8 = true
	o.AllowPartial = true
	o.EmitUnknown = true
	b, _ := o.Marshal(m)
	return string(b)
}
```

Reply:
Good idea, it may reduce hash calculation. Now hash is only calculated when pkpConf exists, if `pkpConf` is nil, empty hash string is used.